pkg/console/subresource/managedserviceproxyresolver/apply.go

```diff
@@ -0,0 +1,48 @@
+package managedproxyserviceresolver
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	clusterproxy "open-cluster-management.io/cluster-proxy/pkg/apis/proxy/v1alpha1"
+	proxyclient "open-cluster-management.io/cluster-proxy/pkg/generated/clientset/versioned/typed/proxy/v1alpha1"
+)
+
+func ApplyManagedProxyServiceResolver(ctx context.Context, client proxyclient.ManagedProxyServiceResolverInterface, required *clusterproxy.ManagedProxyServiceResolver) error {
+	existing, err := client.Get(ctx, required.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err = client.Create(ctx, required, metav1.CreateOptions{})
```

Comment:
If we need to create the resource, we either need. to keep it or get it again, otherwise the variable will contain `nil`
```suggestion
		existing, err = client.Create(ctx, required, metav1.CreateOptions{})
```

Reply:
We shouldn't need to do anything with the result of this create call. If successful, that means the required resource was created and we don't need to do anything more.